llvm/lib/Target/AArch64/AArch64.td

```diff
@@ -554,7 +554,7 @@ def FeatureSME_LUTv2 : SubtargetFeature<"sme-lutv2", "HasSME_LUTv2", "true",
   "Enable Scalable Matrix Extension (SME) LUTv2 instructions (FEAT_SME_LUTv2)">;
 
 def FeatureSMEF8F16 : SubtargetFeature<"sme-f8f16", "HasSMEF8F16", "true",
-  "Enable Scalable Matrix Extension (SME) F8F16 instructions(FEAT_SME_F8F16)", [FeatureSME2, FeatureFP8]>;
+  "Enable Scalable Matrix Extension (SME) F8F16 instructions(FEAT_SME_F8F16)", [FeatureSME2, FeatureFP8, FeatureSMEF16F16]>;
```

Comment:
Likewise, `FeatureSMEF16F16` ought to imply `FeatureSME2` and `FeatureSME` can be removed here.

The idea is to not specify explicitly dependencies that can be transitively derived - 
if you have `A -> B -> C -> D` better to write `A -> B`, `B -> C`, and `C -> D` instead of
`A -> B, C, D`, `B -> C, D`, `C -> D`.

Reply:
The description of `FeatureSMEF16F16` still says
> "Enable SME2.1 non-widening Float16 instructions (FEAT_SME_F16F16)"